Bio/SeqFeature.py

```diff
@@ -384,18 +373,37 @@ def translate(self, parent_sequence, table="Standard", stop_symbol="*",
         >>> f.qualifiers['transl_table'] = [11]
         >>> f.translate(seq)
         Seq('GYTYR*CL**', HasStopCodon(ExtendedIUPACProtein(), '*'))
+
+        Now use the start_offset argument to change the frame. Note
+        this uses python 0-based numbering
+
+        >>> f.translate(seq, start_offset=1)
+        Seq('VTLTDNVSD', ExtendedIUPACProtein())
+
+        Alternatively use the codon_start qualifier to do the same
+        thing. Note: this uses 1-based numbering, which is found
+        in files from NCBI
+
         >>> f.qualifiers['codon_start'] = [2]
         >>> f.translate(seq)
         Seq('VTLTDNVSD', ExtendedIUPACProtein())
         """
         # see if this feature should be translated in a different
         # frame using the "codon_start" qualifier
-        if codon_start not in [1, 2, 3]:
-            raise ValueError("The codon_start must be 1, 2, or 3. The supplied value"
-                    " is {}".format(codon_start))
-        codon_start = int(self.qualifiers.get("codon_start", [codon_start])[0]) - 1
+        try:
+            start_offset = int(self.qualifiers["codon_start"][0]) - 1
+        except KeyError:
+            if start_offset not in [0, 1, 2]:
```

Comment:
This looks wrong, if you have a ``KeyError`` surely ``start_offset`` will not have been defined yet?

Reply:
I think it would be safer to have ``start_offset=None`` as the argument default, meaning use the annotation and otherwise default to zero. Something like:

```python
if start_offset is None:
    try:
        start_offset = int(self.qualifiers["codon_start"][0]) - 1
    except KeyError:
        start_offset = 0  # default value
```